mempool/src/feerate.rs

```diff
@@ -0,0 +1,54 @@
+use common::primitives::amount::Amount;
+
+lazy_static::lazy_static! {
+    pub(crate) static ref INCREMENTAL_RELAY_FEE_RATE: FeeRate = FeeRate::new(1000);
+    pub(crate) static ref INCREMENTAL_RELAY_THRESHOLD: FeeRate = FeeRate::new(500);
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
+pub(crate) struct FeeRate {
+    tokens_per_kb: u128,
+}
+
+impl FeeRate {
+    pub(crate) fn new(tokens_per_kb: u128) -> Self {
+        Self { tokens_per_kb }
+    }
+
+    pub(crate) fn of_tx(fee: Amount, tx_size: usize) -> Self {
+        let tokens_per_byte = Self::div_up(fee.into(), tx_size);
+        Self {
+            tokens_per_kb: tokens_per_byte * 1000,
+        }
+    }
+
+    pub(crate) fn compute_fee(&self, size: usize) -> Amount {
+        Amount::from(self.tokens_per_kb * u128::try_from(size).unwrap() / 1000)
+    }
+
+    pub(crate) fn tokens_per_kb(&self) -> u128 {
+        self.tokens_per_kb
+    }
+
+    fn div_up(fee: u128, tx_size: usize) -> u128 {
+        let tx_size = u128::try_from(tx_size).unwrap();
+        (fee + tx_size - 1) / tx_size
+    }
```

Comment:
Out of curiosity, why are we rounding the result up?

Reply:
Regarding the conversion, I don't think `usize` is ever going to be larger than `u128`. If you want a check for it, it should probably be a static assertion of some sort, as the sizes are known at compilation time when we know the target architecture.

Reply:
The decision of whether to round up or down seemed to me to be arbitrary. In the end I decided to round up in order to be more restrictive (this function is used to compute the minimum rolling fee which is a barrier to entry into the mempool).

Reply:
Regarding the conversion - even if we add a static assert (and that would require using the `static_assertions` crate, which means adding another dependency), we'll still need the conversion, which means we still need to `expect`.

Reply:
Unfortunately, rust doesn't have a decltype like C++... so getting the type of something to know its size isn't really possible. Maybe there's a trick I don't know, and if there's one, let's do it definitely. Static asserts are wonderful and should always be used when possible.

Reply:
> Regarding the conversion - even if we add a static assert (and that would require using the `static_assertions` crate, which means adding another dependency),

Ok, then just have test coverage for extreme values, like passing `usize::MAX` as `tx_size` to see the conversion does not fail in this case.

> we'll still need the conversion, which means we still need to `expect`.

True. I just suggest some sort of extra check the `expect` is justified.

Reply:
Added a test for `div_up`